core/src/main/java/hudson/ClassicPluginStrategy.java

```diff
@@ -206,17 +207,11 @@ private static Manifest loadLinkedManifest(File archive) throws IOException {
             if (libs != null)
                 paths.addAll(Arrays.asList(libs));
 
-            try {
-                Class pathJDK7 = Class.forName("java.nio.file.Path");
-                Object toPath = File.class.getMethod("toPath").invoke(expandDir);
-                URI uri = (URI) pathJDK7.getMethod("toUri").invoke(toPath);
+            Path toPath = expandDir.toPath();
```

Comment:
I thought this was strange when I first saw it...

[`expandDir.toURI().toURL()`](https://docs.oracle.com/javase/7/docs/api/java/io/File.html#toURI()) has been possible since java 1.4?